CHANGELOG.md

```diff
@@ -26,6 +26,12 @@ A brief description of the categories of changes:
 
 ### Changed
 * (gazelle): Update error messages when unable to resolve a dependency to be more human-friendly.
+* (rules) The `PYTHONSAFEPATH` environment variable is no longer used to enable safe-path mode
+  since it is unintentionally inherited by subprocesses. Instead, `-P` is passed to the runtime
+  interpreter given its version can be detected and is greater than or equal to 3.11. If the 
+  runtime interpreter can not be detected or the version is less than 3.11, safe-path mode is not
+  enabled.
```

Comment:
This will be only true for bazel 7 and above because for bazel 6 the starlark implementation of the rules is not used. Maybe worth pointing this out in the changelog.

Reply:
>  for bazel 6 the starlark implementation of the rules is not used

Just for clarification, do you mean that `py_binary` and `py_library` are using the inbuilt Bazel implementations in versions below Bazel 7?

Reply:
If I remember correctly, yes. It is possible to specify the `rules_python` template to be used in bazel 6, but I am not sure how feasible it is to do.

Reply:
Perhaps we can revise the wording here after which I can push a commit:

* (rules) In Bazel 7 and above, the `PYTHONSAFEPATH` environment variable is no longer used to enable safe-path mode since it was unintentionally inherited by subprocesses. Instead, `-P` is passed to the runtime interpreter given its version can be detected and is greater than or equal to 3.11. If the runtime interpreter can not be detected or the version is less than 3.11, safe-path mode is not enabled. Bazel 6 and below do not support the safe-path feature.

Do you think that is concise enough?

Reply:
> Bazel 6 and bellow will continue using the embedded python_bootstrap_template.txt from the bazel repository and are unnafected by the change.

could be better wording.

That said I would like @rickeylev to correct me here if I am misremembering the state of bazel 6 and the python bootstrap template usage.